src/components/CustomFieldForm.tsx

```diff
@@ -0,0 +1,126 @@
+  const formik = useFormik<FormValues>({
+    enableReinitialize: true,
+    initialValues: initialValuesFormik,
+
+    onSubmit: () => {
+      setIsSend(true);
+      const newObj = createObject(formik.values);
+      const dataToServer = JSON.stringify(newObj);
+
+      sendObj(dataToServer).then((res) => setIsValidState(res));
+    },
+    validationSchema: yupSchemaCustomForm,
+  });
+
+  return (
+    <Box sx={{ margin: '35px' }}>
+      <form onSubmit={formik.handleSubmit}>
+        <SingleInputField<FormValues>
+          accessor='dishName'
+          formik={formik}
+          type={'text'}
+          min={'false'}
+          step={'false'}
+          text={'dish name'}
+          valueAsString={true}
+          max={'false'}
+        />
+
+        <SingleInputField<FormValues>
+          accessor='preparationTime'
+          formik={formik}
+          type={'text'}
+          min={1}
+          step={1}
+          text={'preparation time'}
+          max={'false'}
+          valueAsString={true}
+        />
+
+        <SingleSelectInput
+          accessor={'dishType'}
+          formik={formik}
+          text={'dish type:'}
+          arrayValue={dishType}
+        />
+        {formik.values.dishType === 'pizza' && (
+          <>
+            <p>PIZZA</p>
+            <SingleInputField<FormValues>
+              accessor='noOfSlices'
+              formik={formik}
+              type={'number'}
+              min={1}
+              step={1}
+              text={'select number of slices'}
+              valueAsString={true}
+              max={'false'}
+            />
+            <SingleInputField<FormValues>
+              accessor='diameter'
+              formik={formik}
+              type={'text'}
+              min={1}
+              step={'false'}
+              text={'write diameter'}
+              valueAsString={false}
+              max={'false'}
+            />
+          </>
+        )}
+        {formik.values.dishType === 'soup' && (
+          <>
+            <p>SOUP</p>
+            <SingleInputField<FormValues>
+              accessor='spicinessScale'
+              formik={formik}
+              type={'number'}
+              min={1}
+              step={1}
+              text={'write spiciness scale'}
+              valueAsString={false}
+              max={10}
+            />
+          </>
+        )}
+        {formik.values.dishType === 'sandwich' && (
+          <>
+            <p>SANDWICH</p>
+            <SingleInputField<FormValues>
+              accessor='slicesOfBread'
+              formik={formik}
+              type={'number'}
+              min={1}
+              step={1}
+              text={'write number slices of bread'}
+              valueAsString={false}
+              max={'false'}
+            />
+          </>
+        )}
+        <Button type='submit'>submit</Button>
+      </form>
+      {iSsend && isValidState.value === 'isLoading' && <p>isLoading ok</p>}
```

Comment:
formik ma też pola isSending/isPending/isValidating/isError czy jakieś takie podobne sprawdź czy tego się nie da tutaj użyć

Reply:
w przykladach z neta formik troche inaczej wyglada, nie wiem jak to ogarnac na nasze potrzeby

Reply:
tutaj nie dales mi odpowiedzi